src/components/MonthCalendar.tsx

```diff
@@ -8,13 +8,28 @@ interface MonthCalendarProps {
     events: ZetkinEvent[];
 }
 
-const MonthCalendar = ({ month = new Date(Date.now()).getMonth(), events, year = new Date(Date.now()).getFullYear() }: MonthCalendarProps): JSX.Element => {
+const MonthCalendar = ({ month, events, year }: MonthCalendarProps): JSX.Element => {
 
-    const totalDaysInMonth = +new Date(year, 1 + month, 0).getDate();
+    if (!month) month = new Date(Date.now()).getMonth();
+    if (!year) year = new Date(Date.now()).getFullYear();
```

Comment:
Do you know about the `||=` operator? It's new in Typescript 4 but exists in several other languages, and follows the pattern of the established `+=`, `-=`, `*=` etc operators.

Just like `a += 2` is shorthand for `a = a + 2`, this new operator `a ||= 2` is equivalent to `a = a || 2`, i.e. "a equals it's current value (no change) _or_ if falsy, equals 2". Using that operator, these lines could instead read:

```typescript
month ||= new Date(Date.now()).getMonth();
year ||= new Date(Date.now()).getFullYear();
```

Not all coders like this operator, but I do. So I just thought I'd give it some publicity. :)

Reply:
It should be pointed out that all of this only works because in javascript/typescript, the `||` operator returns one of the two operands, not necessarily `true` or `false`.

The statement `a || b` evaluates to either `a` or `b`. It evaluates to `a` if it's truthy, or if it's falsy, it evaluates to `b`. It does not actually care about what `b` is, just return it as it's evaluation.

This can be confusing to some, because the `||` operator is mostly used in strictly boolean operations like if statements and other conditionals, and in those cases the operands are often boolean statements that evaluate to booleans themselves, e.g. `(a < 5) || (a > 500)` which may evaluate to `true || false`, `false || true` or `false || false` depending on the value of `a`.

But the point here is that it's not the `||` operator that's converting anything to a boolean, it's the operand statements themselves (e.g. `a < 5`) that will evaluate to booleans.

If instead you do `a || 5`, it's not going to convert the two operands to booleans. It will however evaluate the "truthyness" of the first operator. If it's truthy, the result is that truthy value. If the value of the first operand is falsy, the result is the other operand.

```js
const x = 5 || 25;
console.log(x);
```

This will print out `5` (the first operand, because it's truthy), not `true`.

```js
const x = false || 0;
console.log(x);
```

This will print out `0` (the second operand, because the first is falsy), not `false`.

A curious property of hos the `||` operator works in javascript is that if the first operand is truthy, the operator will definitely evaluate to that operand, so it doesn't even need to evaluate the second one. You can see this in action if the second one is a function:

```javascript
const foo = () => console.log('bar!');
const x = 2 || foo(); // x is now 2, nothing is printed
const y = 0 || foo(); // foo() is invoked, "bar!" is printed
                      // y is now undefined (return value of foo())
```

That's a great property in this context, because it means that just like the `if (!month)` you currently have, the `month ||=` solution (or long-form `month = month || …`) will not cause the `Date` constructor to be executed unnecessarily.

Reply:
cool!